internal/gen/input.go

```diff
@@ -368,30 +371,64 @@ func (dg *dataGenerator) Do(f func(tbl flux.Table) error) error {
 		sg.Type = vt.Type
 
 		for _, s := range sg.Series {
-			builder := execute.NewColListTableBuilder(s, dg.Allocator)
-			startIdx, _ := builder.AddCol(flux.ColMeta{
-				Label: execute.DefaultStartColLabel,
-				Type:  flux.TTime,
+			tbl, err := tableutil.Stream(func(ctx context.Context, fn tableutil.SendFunc) error {
+				// Construct the table columns.
+				cols := make([]flux.ColMeta, len(s.Cols())+2)
+				copy(cols, s.Cols())
+				cols[len(cols)-2] = flux.ColMeta{Label: execute.DefaultTimeColLabel, Type: flux.TTime}
+				cols[len(cols)-1] = flux.ColMeta{Label: execute.DefaultValueColLabel, Type: sg.Type}
+
+				// Only construct the key values once for the first table
+				// and then reuse them for each one with a slice.
+				// The first table should always be the biggest because of
+				// the size constraint.
+				// These are constructed lazily below.
+				keyValues := make([]array.Interface, len(s.Cols()))
+				defer func() {
+					for _, vs := range keyValues {
+						if vs != nil {
+							vs.Release()
+						}
```

Comment:
Question: why there is no Release like this in `TestStream`?

Reply:
It's because this piece of code is being clever. Because of the uniform nature of how this constructs tables, it knows that the first buffer will always be the maximum size. So when it constructs group key columns, it attempts to construct them as optimally as possible. It then keeps a reference here and retains a new copy to use for the table buffer below. But, since it is maintaining its own reference, it has to release it when it is done.

So if you generate `1024 * 10` column readers, it only generates one buffer's worth of key values and just shares it between the 10 column readers.